packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts

```diff
@@ -62,8 +62,7 @@ export interface DateRangeCalendarSlotProps<TDate extends PickerValidDate>
 
 export interface ExportedDateRangeCalendarProps<TDate extends PickerValidDate>
   extends ExportedDayCalendarProps<TDate>,
-    BaseDateValidationProps<TDate>,
```

Comment:
The `BaseDateValidationProps` is still imported in this file because it's used to define `DateCalendarDefaultizedProps`

```ts
export type DateRangeCalendarDefaultizedProps<TDate extends PickerValidDate> = DefaultizedProps<
  DateRangeCalendarProps<TDate>,
  | 'views'
  | 'openTo'
  | 'reduceAnimations'
  | 'calendars'
  | 'disableDragEditing'
  | 'availableRangePositions'
  | keyof BaseDateValidationProps<TDate>
>;
```

Would it make sense to have `type ExportedValidateDateRangePropsDefaultizeKey = keyof BaseDateValidationProps<TDate>`. to completely remove links with those notion of `BaseDateValidationProps`, `BaseTimeValidationProps`, ...

Seems their is already something similar with `ExportedValidateTimeRangeProps` and `ValidateTimeRangeProps`. The second being the defaultized version of the first

Reply:
That's an interesting point.
In #14679 I'm reworking how we apply the default values (and it will remove interfaces like `DateRangeCalendarDefaultizedProps`), but I'm just moving the logic elsewhere (see https://github.com/mui/mui-x/pull/14679/files#diff-9b39de78b5e067c04513d22bde87e071d6adc77dc03839437ca392b977fc859dR42).
We could indeed introduce new interfaces next to the validator, or maybe even compare `ExportedXXX` and `XXX` and automatically list those which are optional in `ExportedXXX` but required in `XXX` :thinking: 

I'll have a look in a follow up :+1: